pkg/apis/autoscaling/v1alpha1/pa_defaults.go

```diff
@@ -17,8 +17,19 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"strconv"
+
 	"github.com/knative/serving/pkg/apis/autoscaling"
 	servingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/knative/serving/pkg/autoscaler/config"
```

Comment:
I'm still a bit uncomfortable with the direction of this dependency, even with the cycle broken.

Part of me wonders whether this config should simply move under `./pkg/apis/autoscaling`, but it contains a ton of knobs specific to our implementations, which also feels off.

Remind me why we are defaulting class-specific annotations in the webhook instead of just doing so in the controller?  This probably belongs in a comment on the switch below.

Reply:
We were defaulting the class in the webhook to make sure that there was one.  Seemed like the right place at the time.  We also defaulted the metric.

When introducing the window annotation, I didn't want to change the behavior of existing Revisions (backwards compatibility, safety and all that).  And when the defaults are changed in the config map, I don't want them rolled out to all Revisions immediately.  So I decided to write them down on the PodAutoscaler as a snapshot of config when the Revision was created.  Webhook seemed like the right place to do that.

However I could move the defaulting to the controller.  What are the downsides?

Reply:
I don't feel strongly one way or the other.  If I move config back out of the webhook and default in the controller, then the existing default values in the webhook (e.g. class) will have to move too when I implement default class in the config map (https://github.com/knative/serving/issues/3415).

@mattmoor if you feel strongly to keep the dependencies of the types package separate from config map and their implementation details, I'll undo the config move.  No problem.  Sound good?

Reply:
Yeah, I think I'd like to see us plumb it into the webhook without globals, and consider the config for defaulting holistically.  We need this for all resource classes, and probably for several default values in our public API.

Reply:
I've removed the part of this pr which moves config into the webhook.  We just default in the controller now.